### ISSUE TITLE
Improve parsing error recovery on an implicit array creation with a length

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#24701]
It currently produces various syntax errors.
```cs
public class C {
    public void M() {
        var x = new [3] { 1, 2, 3 }
    }
}
```
```
error CS1003: Syntax error, ']' expected
error CS1514: { expected
error CS1003: Syntax error, ',' expected
error CS1003: Syntax error, ',' expected
error CS1002: ; expected
error CS1513: } expected
error CS0623: Array initializers can only be used in a variable or field initializer. Try using a new expression instead.
```

**Comment by jcouv** [comment]
Linking to https://github.com/dotnet/roslyn/issues/24699

**Comment by jcouv** [comment]
Fixed in 15.7. Closing.
